.devcontainer/Dockerfile

```diff
@@ -1,9 +1,55 @@
 FROM ubuntu:22.04 AS base
 
-WORKDIR /home/
+## Add non-root user for startup of pg-embed with rust backend
 
-COPY . .
+ARG USERNAME=user
 
-RUN bash ./setup.sh
+RUN useradd -m $USERNAME && usermod -aG sudo $USERNAME
+RUN echo "user:pass" | chpasswd
 
-ENV PATH="/root/.cargo/bin:$PATH"
+## Install common dependencies
+
+RUN apt-get update && \
+   apt-get install -y \
+   curl \
+   git \
+   gnupg2 \
+   sudo \
+   build-essential \
+   openssl \
+   cmake \
+   pkg-config \
+   libssl-dev
+
+## Setup Java and Python
+
+RUN apt-get install -y \
+   gcc clang libclang-dev python3-pip python3-plumbum \
+   hub numactl openjdk-19-jre-headless maven
+
+## Install nodejs and global packages
+
+RUN curl -fsSL https://deb.nodesource.com/setup_19.x | bash -
+RUN apt-get install -y nodejs
+RUN npm install --global yarn
+RUN npm install --global openapi-typescript-codegen
+
+# Switch to non-root user
+USER $USERNAME
+
+## Install rustup and common components
+
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
+ENV PATH="/root/.cargo/bin:/root/.cargo/env:${PATH}"
+
+SHELL ["/bin/bash", "-c"]
+RUN source "$HOME/.cargo/env" && \
+   rustup install nightly && \
+   rustup component add rustfmt && \
+   rustup component add rustfmt --toolchain nightly && \
+   rustup component add clippy && \
+   rustup component add clippy --toolchain nightly
+SHELL ["/bin/sh", "-c"]
+
+# Set locale to fix pg-embed startup
+ENV LC_ALL=C
```

Comment:
I'm not sure if LC_ALL=C is the right encoding actually. Maybe something more specific e.g., UTF-8, is better as the  strings we store in the db will be in UTF-8. 

I remember that this caused problems in the past on windows e.g., https://github.com/feldera/dbsp/commit/56f72e37fbcaca46d30f8c8f1391b74199914ac0

Maybe @blp has a suggestion?